pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -505,6 +505,10 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 
 	pod := c.pods.getPodByIP(proxyIP)
 	if pod != nil {
+		// for k8s multi cluster, in case there are pods of the same ip across clusters
+		if proxy.ID != pod.Name+"."+pod.Namespace {
```

Comment:
Can we add more comment here?  when there is same ip exist in 1+ clusters, how are we handling it?

Reply:
sgtm

Reply:
done

Reply:
Now that I think of it, isn't there still a possible corner case. What if `helloworld:v1` is running in both clusters? Is it not possible in that case that there could be a pod in cluster2 with the same pod name and namespace as one in cluster1? Seems to me that the proxy.ID check isn't guaranteed correct. Is there a way to check if the pods cluster is the same?

Reply:
YEah pod names can also be specified by end user.. So this fix wont work. I recall that in the VPN multicluster case, there is a strict requirement that the pod CIDRs *must* be different. If they are the same, the system wont work as guaranteed. Now, if this is a problem for the non VPN, split horizon case, then we cannot do much unfortunately. You could add node metadata to indicate which cluster the pod is coming from.. So that you can distinguish based on the node metadata field, plus the pod's IP. But thats still a brittle fix IMO.
Btw, this is one of the problems with any solution (split horizon and vpn) that wants to see all endpoints in another cluster.

Reply:
helloworld:v1 is expected to run in both clusters. 

And we already have a solution we agreed on - in split horizon case, we use 'network' ( which is included in the metadata so visible to pilot, and included as a label ) to identify the network. And inside a network, as well as for VPN, we assume proper non-overlapping IP allocation.

Reply:
So that last part `we assume proper non-overlapping IP allocation` is not documented in the split-horizon example. If that is an additional restriction that we are requiring, we should add that to the documentation. It means that split-horizon is not a usable in environments that don't guarantee non-overlapping IPs in the clusters. That includes IBM Cloud and wherever @hzxuzhonghu was trying it.

Reply:
>What if helloworld:v1 is running in both clusters? Is it not possible in that case that there could be a pod in cluster2 with the same pod name and namespace as one in cluster1? Seems to me that the proxy.ID check isn't guaranteed correct. Is there a way to check if the pods cluster is the same?


Yes, this possibility is very very low(podname.podnamespace and podIP both same).   Currently we can not get the proxy clusterID, only can get the pods'.